models/Freelance.js

```diff
@@ -79,6 +79,28 @@ Freelance.pre('save', function (next) {
 	next();
 
 });
+//call of asynchronous, parallel pre
+// method 'validate' is not executed until done() is called.
+// validate is also called for every update, not only save.
+Freelance.pre('validate', true, function(next, done){
+	var count = 0;
+	next();
+	if (this.reviews.length == 0) done();
+	for (var item of this.reviews){
+		Review.findById(item).exec((err, review)=>{
+			if (err) throw err;
+			if (review){
+				this.avgScore += review.score;
+				count++;
+			}
+			if (count >= this.reviews.length) { 
+				this.avgScore = Math.round(this.avgScore/this.reviews.length); // different possibilities for rounding: floor, ceil
```

Comment:
I thought for the display it was more pleasing for the front end to have integers. (that way it is coherent with the box selection given to the user)
I can take it out but i suggest front-end then to round it before showing it